packages/vite/src/node/config.ts

```diff
@@ -242,10 +244,9 @@ export interface SharedEnvironmentOptions {
    */
   consumer?: 'client' | 'server'
   /**
-   * Runtime Compatibility
-   * Temporal options, we should remove these in favor of fine-grained control
+   * Don't replace `process.env` to a static value.
```

Comment:
```suggestion
   * If true, `process.env` referenced in code will be preserved as-is and evaluated in runtime.
   * Otherwise, it is statically replaced as an empty object.
```

Perhaps this can this describe the behaviour more specifically.